In [2]:
import pandas as pd

In [4]:
titanic_df = pd.read_csv("titanic_data.csv")

In [5]:
titanic_df.shape

(891, 12)

In [11]:
titanic_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


Specific Information

In [8]:
# Star info : mean 01 medium(Q2) Q3 mode std min max corr
titanic_df[["Age", "Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [9]:
titanic_df[["Age", "Fare"]].median()

Age     28.0000
Fare    14.4542
dtype: float64

In [10]:
titanic_df[["Age", 'Fare']].describe()

,Age,Fare
count,714.000000,891.000000
mean,29.699118,32.204208
std,14.526497,49.693429
min,0.420000,0.000000
25%,20.125000,7.910400
50%,28.000000,14.454200
75%,38.000000,31.000000
max,80.000000,512.329200


In [12]:
titanic_df.Embarked.mode()

0    S
Name: Embarked, dtype: str

Age
The average age of passengers was about 30 years.
The middle age was 28 years.
The youngest passenger was 0.42 years, and the oldest was 80 years.
Most passengers were between about 20 and 38 years old.
The mean and median are close, so the age data is fairly balanced.
Fare
The average fare was about 32.20.
The middle fare was 14.45.
The cheapest fare was 0, and the most expensive was 512.33.
Most passengers paid between about 8 and 31.
The mean is much higher than the median because a few passengers paid very expensive fares.


In [ ]:
# Counts
# titanic_df.value_counts("Embarked" 

In [ ]:
#Emabraked form S, Fare Distribution
titanic_df[titanic_df.Embarked == "S"]["Fare"].describe()

count    644.000000
mean      27.079812
std       35.887993
min        0.000000
25%        8.050000
50%       13.000000
75%       27.900000
max      263.000000
Name: Fare, dtype: float64

In [1]:
#Embarked from S anf gender Male, Survival rate ?

titanic_df[
    (titanic_df.Embarked == "S") & (titanic_df.Sex == "Male")
].value_counts("Survived")

NameError: name 'titanic_df' is not defined